modules/brainbrowser/js/brainbrowser.loris.js

```diff
@@ -750,15 +750,37 @@ $(function() {
 
     for (i = 0; i < minc_ids_arr.length; i += 1) {
 
-        minc_volumes.push({
-            type: 'minc',
-	    header_url: "",
-	    raw_data_url: loris.BaseURL + "/brainbrowser/ajax/minc.php?minc_id=" + minc_ids_arr[i],
-            template: {
-                element_id: "volume-ui-template4d",
-                viewer_insert_class: "volume-viewer-display"
-            }
+        var filename = null;
+        $.ajax({
+            url: loris.BaseURL + "/brainbrowser/ajax/getMincName.php",
+            data: 'minc_id=' + minc_ids_arr[i],
+            method: 'POST',
+            success: function (data) {
+              filename = data;
+            },
+            async: false
```

Comment:
If all you're trying to do is get rid of the async, I think you could replace the `async: false` with a promise chain by using `.then` on the ajax call and putting the `if/else if` inside of a function inside of the `then`.

ie.

```
$.ajax({ ... filename url stuff, minus async}).then(function(filename) {
     if (filename.endsWith("mnc") { ... } else if { ... } 
     etc
});
```

Alternatively (and probably better..), the if the script at`ajax/minc.php?minc_id=...` can be modified to include a `Content-Type` header, you could use that to push the appropriate object to BrainBrowser's volumes based on the header in the reply and only have to do one request.

Reply:
@driusan Is it not necessary to instantiate a promise object when using an ajax call?

Reply:
@HenriRabalais jquery ajax calls are (allegedly, I haven't tried) promise compatible, so it shouldn't be necessary.